xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * An {@link XdsClient} instance encapsulates all of the logic for communicating with the xDS
+ * server. It may create multiple RPC streams for a series of xDS protocols (e.g., LDS, RDS,
+ * VHDS, CDS and EDS) over a single channel. Watch-based interfaces are provided for each set
+ * of data needed by gRPC.
+ *
+ * <p>This class should only be instantiated by the xDS resolver but can be passed to load
+ * balancing policies.
+ */
+abstract class XdsClient {
```

Comment:
Do you expect subclasses of `XdsClient`? Instead of `abstract` it should be `final` right? Also can it be made`public` so classes in child packages (e.g. `sds`) can access it?

Reply:
I expect `XdsClient` will be quite complicated with a lot of dependencies on xDS protocol, so make it `abstract` for testing purposes. Also, I did not consider the use case for `sds`. But yes, any concerns are welcome as `XdsClient` will be the joint for many pieces, reconciling the integration early is better than late.